### DISCUSSION TITLE
Add a preference to automatically generate Taxon GUIDs

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5895] [Feature Requests]
> Is there a way to have the system create GUIDs for taxa as they are created, otherwise we will have to regularly generate GUIDs for those taxa without them.

**Requested by:** RBGE

**Comment by melton-jason** [comment]
What is the reasoning for currently not generating and assigning Taxon and Geography GUIDs? 

GUIDs as of now are set by the backend as a result of the following snippet: 
https://github.com/specify/specify7/blob/2624e73f6b37c8eec9ab079b310611a760378bbd/specifyweb/businessrules/guid_rules.py#L6-L11

**Comment by maxpatiiuk** [comment]
The reasoning (and that decision was originally made 10 or 20 years ago in sp6 and carried over into sp7) is that people would assign their own GUIDs OR they would get data from some source and use that source's GUIDs.

i.e specify is not a Taxa management system, so we are not in the business of assigning GUIDs for Taxa (same for Geography)
we are in the business of Collection Object management, so we are assigning GUIDs for that

though that is a bit idealistic as we don't provide easy enough ways to get taxa authority data into specify, and we definitely don't have an easy way to update the taxa with updates in the taxa authority or to even keep them in sync.

so I suppose this decision was part of us being a bit more opinionated/trying to enforce the practices we belive are good - but in this case we didn't go all the way to actually make those good practices easy on people.

so yeah, I would be ok with adding a global pref for that (AND WE NEED INSTITUTIONAL PREFERENCES - pretty urgently - so many features are held back because of lack of that)

**Comment by maxpatiiuk** [comment]
@melton-jason Regardless of the permanent solution, in the short term, could you add a one-line comment above that  `if sender in (Taxon, Geography): return` line (could be just as simple as `# see <link to the comment I posted above>`)

thanks

**Comment by emenslin** [comment]
From #2081 

> Also for the Geography tree. This was intentionally decided, but maybe we should consider that the generated GUIDs can be overwritten in the future if so desired anyway.
> 
> I think it would be nice if we could use any field as a GUID. Maybe any string field could have GUID generated in it if the form definition specified it? Is that possible?
